pkg/services/libraryelements/libraryelements.go

```diff
@@ -83,6 +83,16 @@ func (l *LibraryElementService) DisconnectElementsFromDashboard(c context.Contex
 
 // DeleteLibraryElementsInFolder deletes all elements for a specific folder.
 func (l *LibraryElementService) DeleteLibraryElementsInFolder(c context.Context, signedInUser identity.Requester, folderUID string) error {
+	// first check that the folder exists
+	_, err := l.folderService.Get(c, &folder.GetFolderQuery{
```

Comment:
This change is to keep the same functionality we have today. `deleteLibraryElementsInFolderUID` was using the function `requireEditPermissionsOnFolderUID`, which was using the `NewByUID` guardian. In that guardian, it was first checking that the folder existed via the uid. This was the only usage of a folder using `NewByUID`.

We cannot just add this to the folder guardian, as there are legitimate cases where the user cannot read the parent folder but can read the item inside.